### DISCUSSION TITLE
【评论 插件】valine评论跨域、waline部署后评论时报错但能正常评论上

### DISCUSSION THREAD

**Discussion by noshore** [tangly1024/NotionNext#2049] [General]
**描述遇到的问题**
waline通过vercel部署之后评论提交弹出提示Cannot read properties of undefined (reading ‘name’)
但是刷新之后评论是评论上的 waline后台也有显示 请问博主大大这个是因为什么

**相应配置**
blog.noshore.cn

**截图**
相关的页面，应该用结果

**环境**

- 操作系统: windows
- 浏览器 chrome
- 版本 [e.g. 22]

**Comment by tangly1024** [comment]
似乎是vercel 版本的 waline 还有一些bug ，后续我会跟进解决这个问题。

目前我暂时用的Deta方案进行的部署。

**Comment by noshore** [comment]
> 似乎是vercel 版本的 waline 还有一些bug ，后续我会跟进解决这个问题。
> 
> 目前我暂时用的Deta方案进行的部署。

大佬 deta的方案好像也出问题了 部署会显示没有这个项目
<a href="https://imgse.com/i/zK2wPe"><img src="https://s1.ax1x.com/2022/11/20/zK2wPe.png" alt="zK2wPe.png" border="0" /></a>

**Comment by tangly1024** [comment]
诶，我刚刚试了一下还是成功的：
<img width="591" alt="图片" src="https://user-images.githubusercontent.com/15920488/202885372-2136e905-0da6-4121-bdaa-6cb90d243c1d.png">


或者你可以尝试一下 Railway的部署方案：
https://www.dsuper.xyz/article/waline-setting

**Comment by noshore** [comment]
> 诶，我刚刚试了一下还是成功的： <img alt="图片" width="591" src="https://user-images.githubusercontent.com/15920488/202885372-2136e905-0da6-4121-bdaa-6cb90d243c1d.png">
> 
> 或者你可以尝试一下 Railway的部署方案： https://www.dsuper.xyz/article/waline-setting

尴尬 找到问题所在了 deta注册验证完我直接点了大佬博客里的一键部署链接 deta需要先选择数据中心创建项目 选后之后点一键部署就可以了

**Comment by noshore** [comment]
> 诶，我刚刚试了一下还是成功的： <img alt="图片" width="591" src="https://user-images.githubusercontent.com/15920488/202885372-2136e905-0da6-4121-bdaa-6cb90d243c1d.png">
> 
> 或者你可以尝试一下 Railway的部署方案： https://www.dsuper.xyz/article/waline-setting

大佬 用deta部署好像无法添加变量 这样就不能设置邮件通知了 所以我尝试部署valine 出现的问题就是评论提交之后无法显示并且后台也没有 环境变量应该是没有问题的 是不是由于国际版墙的问题 因为我发现国际版云引擎域名被q
还有一个问题是我在博主你的博客尝试使用valine 但是valine下方的评论不会显示出来也无法评论成功 waline是正常使用的 下方能显示别人的评论 但是搭梯子的话 valine也能显示评论了 我这边江苏电信

**Comment by tangly1024** [comment]
感谢反馈，如您所言，国际版被墙😭😭😭

deta的变量支持很少，这也是我目前头痛的问题

**Comment by noshore** [comment]
> 感谢反馈，如您所言，国际版被墙😭😭😭
> 
> deta的变量支持很少，这也是我目前头痛的问题

博主想请问一下你valine使用的是国内还是国际 我部署的国际版连评论框下方的来发评论吧那一行字都无法正常显示 我想请教一下这个是由于我部署的国际版的原因还是因为我的部署错误（如下图所示） 我感觉是不是我哪里部署错误 因为我的博客中即便是搭了梯子也无法正常评论 而博主你的博客搭了梯子就能和waline同步并且正常评论
博客地址：blog.noshore.cn
<a href="https://imgse.com/i/zQHtCn"><img src="https://s1.ax1x.com/2022/11/21/zQHtCn.png" alt="zQHtCn.png" border="0" /></a>

**Comment by tangly1024** [comment]
我使用的是国际版，看问题你需要检查一下你的Valine地址；

NEXT_PUBLIC_VALINE_SERVER_URLS

## 错误示范
> xxxx.api.lncldglobal.com
## 正确示范

> https://xxxx.api.lncldglobal.com
<img width="985" alt="图片" src="https://user-images.githubusercontent.com/15920488/203067021-9a5cdade-27a1-4b12-bb3e-e0545dd6b331.png">

**Comment by noshore** [comment]
> 我使用的是国际版，看问题你需要检查一下你的Valine地址；
> 
> NEXT_PUBLIC_VALINE_SERVER_URLS
> 
> ## 错误示范
> > xxxx.api.lncldglobal.com
> 
> ## 正确示范
> > https://xxxx.api.lncldglobal.com
> 
> <img alt="图片" width="985" src="https://user-images.githubusercontent.com/15920488/203067021-9a5cdade-27a1-4b12-bb3e-e0545dd6b331.png">



> 我使用的是国际版，看问题你需要检查一下你的Valine地址；
> 
> NEXT_PUBLIC_VALINE_SERVER_URLS
> 
> ## 错误示范
> > xxxx.api.lncldglobal.com
> 
> ## 正确示范
> > https://xxxx.api.lncldglobal.com
> 
> <img alt="图片" width="985" src="https://user-images.githubusercontent.com/15920488/203067021-9a5cdade-27a1-4b12-bb3e-e0545dd6b331.png">

确实是这个问题 不过博主你是如何做到让waline和valine在博客里数据同步的 我的在waline评论之后切换到valine并不显示 但是valine admin后台却是有显示的 请问这个是什么原因导致的

**Comment by tangly1024** [comment]
valine似乎是存在网络跨域的问题 最近我的valine都不显示评论了😢
